vaadin-dashboard-flow-parent/vaadin-dashboard-flow/src/main/java/com/vaadin/flow/component/dashboard/DashboardSection.java

```diff
@@ -0,0 +1,193 @@
+import com.vaadin.flow.component.Tag;
+import com.vaadin.flow.component.dependency.JsModule;
+import com.vaadin.flow.component.dependency.NpmPackage;
+import com.vaadin.flow.dom.Element;
+
+/**
+ * @author Vaadin Ltd
+ */
+@Tag("vaadin-dashboard-section")
+@NpmPackage(value = "@vaadin/polymer-legacy-adapter", version = "24.5.0-alpha8")
+@JsModule("@vaadin/polymer-legacy-adapter/style-modules.js")
+@JsModule("@vaadin/dashboard/src/vaadin-dashboard-section.js")
+// @NpmPackage(value = "@vaadin/dashboard", version = "24.6.0-alpha0")
+public class DashboardSection extends Component implements HasWidgets {
+
+    private final List<DashboardWidget> widgets = new ArrayList<>();
+
+    private final DashboardChildDetachHandler childDetachHandler;
+
+    /**
+     * Creates an empty section.
+     */
+    public DashboardSection() {
+        this(null);
+    }
+
+    /**
+     * Creates an empty section with title.
+     *
+     * @param title
+     *            the title to set
+     */
+    public DashboardSection(String title) {
+        super();
+        childDetachHandler = getChildDetachHandler();
+        setTitle(title);
+    }
+
+    /**
+     * Returns the title of the section.
+     *
+     * @return the {@code sectionTitle} property from the web component
+     */
+    public String getTitle() {
+        return getElement().getProperty("sectionTitle");
+    }
+
+    /**
+     * Sets the title of the section.
+     *
+     * @param title
+     *            the title to set
+     */
+    public void setTitle(String title) {
+        getElement().setProperty("sectionTitle", title);
+    }
+
+    @Override
+    public List<DashboardWidget> getWidgets() {
+        return Collections.unmodifiableList(widgets);
+    }
+
+    @Override
+    public Stream<Component> getChildren() {
+        return widgets.stream().map(Component.class::cast);
+    }
+
+    @Override
+    public void add(DashboardWidget... widgets) {
+        Objects.requireNonNull(widgets, "Widgets to add cannot be null.");
+        List<DashboardWidget> toAdd = new ArrayList<>(widgets.length);
+        for (DashboardWidget widget : widgets) {
+            Objects.requireNonNull(widget, "Widget to add cannot be null.");
+            toAdd.add(widget);
+        }
+        toAdd.forEach(this::doAddWidget);
+        updateClient();
+    }
+
+    @Override
+    public void addWidgetAtIndex(int index, DashboardWidget widget) {
+        Objects.requireNonNull(widget, "Widget to add cannot be null.");
+        if (index < 0) {
+            throw new IllegalArgumentException(
+                    "Cannot add a widget with a negative index.");
+        }
+        if (index > getWidgets().size()) {
+            throw new IllegalArgumentException(String.format(
+                    "Cannot add a widget with index %d when there are %d widgets",
+                    index, getWidgets().size()));
+        }
+        doAddWidgetAtIndex(index, widget);
+        updateClient();
```

Comment:
Most of the `updateClient();` calls are tested (by unit tests, some covered by IT), but I found 4 in `DashboardSection` and 4 in `Dashboard` that aren't.

Reply:
Added more ITs to cover cases that needs client update. I believe most of them should be covered now.